cirq/ops/pauli_string.py

```diff
@@ -70,6 +82,21 @@ def get(self, key: raw_types.Qid, default=None):
     def __mul__(self, other):
         if isinstance(other, (int, float, complex)):
             return PauliString(self._qubit_pauli_map, self._coefficient * other)
+        if isinstance(other, PauliString):
+            s1 = set(self.keys())
+            s2 = set(other.keys())
+            extra_phase = 1
+            terms = {}
+            for c in s1 - s2:
+                terms[c] = self[c]
+            for c in s2 - s1:
+                terms[c] = other[c]
+            for c in s1 & s2:
+                f, p = self[c].phased_pauli_product(other[c])
+                extra_phase *= f
+                if p is not None:
+                    terms[c] = p
+            return PauliString(terms, self.coefficient * extra_phase)
```

Comment:
I think the second argument should be `self.coefficient * other.coefficient * extra_phase`.

Reply:
Fixed and added test to catch it.